scrapy/http/request/__init__.py

```diff
@@ -96,3 +96,31 @@ def replace(self, *args, **kwargs):
             kwargs.setdefault(x, getattr(self, x))
         cls = kwargs.pop('cls', self.__class__)
         return cls(*args, **kwargs)
+
+    def mark_for_retry(self, reason=None, **kwargs):
```

Comment:
I think that `return req.retry()` reads better than `req.mark_for_retry()`.
Besides that, _mark_, for me, reads as something that is changing the instance and not creating a new one.

Reply:
Besides that, I don't know if this behavior should be here.
Should a request know about the configurations other modules need to retry it?
Or should the retrying module set this configurations in the request?

Reply:
I agree with `req.retry()`, which is similar to `req.replace()`.